### PR TITLE
Added an and and an or condition

### PR DESCRIPTION
**Pull request by eneufeld**
* added test cases
* refactored code to cope with the requirement of supporting composite conditions

### CODE REVIEW

File: packages/core/test/util/runtime.test.ts [hunk 1]
```diff
@@ -52,6 +54,130 @@ test('evalVisibility show valid case', t => {
     t.is(evalVisibility(uischema, data), true);
 });
 
+test('evalVisibility show valid case based on andcondition', t => {
```
Comment (edgarmueller) [inline packages/core/test/util/runtime.test.ts]: whitespace missing in test names

File: packages/core/src/util/renderer.ts [hunk 2]
```diff
@@ -309,12 +309,10 @@ export const mapStateToControlProps =
     (state: JsonFormsState, ownProps: OwnPropsOfControl): StatePropsOfControl => {
         const { uischema } = ownProps;
         const path = composeWithUi(uischema, ownProps.path);
-        const rulePath = _.has(uischema, 'rule')
-            && composeWithUi(uischema.rule.condition, ownProps.path);
         const visible = _.has(ownProps, 'visible') ?
-            ownProps.visible : isVisible(ownProps, state, rulePath);
+            ownProps.visible : isVisible(ownProps, state, ownProps.path);
```
[the remark this answers is not in the document]
Comment (eneufeld) [reply packages/core/src/util/renderer.ts]: I'm not 100% sure, but it is used in runtime/getConditionScope when composing the condition path, so I would keep it and we should check whether it makes sense

**Comment by smrgeoinfo** [comment]
where is the documentation explaining how this works?

**Comment by sdirix** [comment]
Hi @smrgeoinfo, this is an outdated approach. Please have a look [here](https://jsonforms.io/docs/uischema/rules) for the "new" generic schema-based approach.

**Pull request merged**